sklearn/preprocessing/tests/test_data.py

```diff
@@ -1367,6 +1367,23 @@ def test_quantile_transform_subsampling():
     # linspace CDF
     assert len(np.unique(inf_norm_arr)) == len(inf_norm_arr)
 
+    # check subsampling disabled
+    n_samples = 1000000
```

Comment:
It would be great if we could come up with a way to check that subsampling was disabled/not used without having to use a large dataset. It might be fairly fast to do five rounds with 1M data points, but because scikit-learn has so many test cases I think it is worth trying to spend a bit of time thinking about an alternative.

For example, could we use a smaller dataset and check that the percentiles computed inside the estimator are equal to `np.nanpercentile` on the same dataset? Basically what you are doing here but with a smaller dataset and maybe only one round? I think the whole process (w/o subsampling) is deterministic so we don't need re-runs.

Reply:
actually your last point makes for a good test, run the procedure twice on a sample set and check that the quantiles are approximately equal. We could do this with a sample that is a bit larger than the default subsample value.
To make the test more robust we could add a few extreme values (e.g. q>0.98) as these will have a noticeable effect and relatively low chance of all of them being sampled.